docs/solidity-to-cadence.mdx

```diff
@@ -0,0 +1,447 @@
+
+## msg.sender considered harmful
+
+The first question that every Solidity developer asks when they start programming in Cadence is:
+
+**"How do I get the account who authorized the transaction?"**
+
+In Ethereum this account is referred to as `msg.sender` and informs the program flow in a function depending on who
+authorized it. Doing so is key to access and security, and is the basis of identity and ownership on Ethereum.
+
+Cadence does not have `msg.sender` and there is no transaction-level way for Cadence contract to uniquely identify the
+calling account. Even if there was a way to access it, Cadence supports [multi-sig](solidity-to-cadence#multi-key-multi-signature-support)
+transactions, meaning that a list of all the signers' accounts would be returned.
+
+The reason `msg.sender` is both unsupported and strongly advised against is because Cadence uses Capabilities for
+access rather than addresses. The mindset change that developers need to adjust to is that a capability must first be
+obtained by the authorizing account (called provider or signer in Cadence) from the contract that will require it,
+which then enables the requesting account to access the protected function or Resource. This means the contract never
+needs to know who the signer is before proceeding because the capability **IS** the authorization.
+
+The [capability-based security](https://en.wikipedia.org/wiki/Capability-based_security) model frames access in
+the opposite direction than the [access-based security](https://en.wikipedia.org/wiki/Access-control_list) model.
+
+<Img src="https://storage.googleapis.com/flow-resources/documentation-assets/solidity-to-cadence/access-based-security.png" />
+<Img src="https://storage.googleapis.com/flow-resources/documentation-assets/solidity-to-cadence/capability-based-security.png" />
+
+## Access control using Capabilities
+
+Solidity lacks specific types or other primitives to aid with permission management. Developers have to inline
+guards to `require` at every function entry-point, thus validating the `msg.sender` of the transaction.
+
+[Capabilities](language#capability-based-access-control) are defined by linking storage paths (namespaces for contract
+storage) to protected objects and then making that linked capability available to other accounts. Public and private
+scopes defined for storage paths and Capabilities themselves align precisely with
+[PublicAccount](language#accounts#publicaccount)/[AuthAccount](language#accounts#authaccount) account scopes.
+
+Any account can get access to an account's public Capabilities. Public capabilities are created using public paths,
+i.e. they have the domain `public`. For example, all accounts have a default public capability linked to the
+`FLOWToken.Vault` Resource. This vault is exposed as a public capability, scoped to the `receiver` resource interface,
+to allow any account to `borrow()` the Vault to make a `deposit()`. Public capabilities can be obtained from both
```

Comment:
```suggestion
to allow any account to `borrow()` a reference to the Vault to make a `deposit()`. Since only the methods defined under the `FungibleToken.Receiver` interface are exposed, whoever borrows the reference to the vault won't be able to call the `withdraw()` method that allows to take tokens out of that Vault. Public capabilities can be obtained from both
```

Reply:
Not 100% sure about the wording here, but it seemed a good place to insist on *how* capabilities make secure to access resources